rxnutils/chem/reaction.py

```diff
@@ -135,6 +135,48 @@ def rinchi_key_short(self) -> str:
             self._build_rinchi_info()
         return self._rinchi_data.short_rinchikey
 
+    def generate_coreagent(self):
+        """Extract un-mapped product atoms as extra ractant fragments"""
+        if not self.has_partial_mapping():
+            return ""
+
+        extra_reactant_fragment = ""
+        for product in self.products:
+            # Get unmapped atoms
+            unmapped_ids = [
```

Comment:
Lines 146-154 are very similar to lines 209-215, maybe we should have a separate function for them?